ndarray/src/main/java/org/tensorflow/ndarray/index/Index.java

```diff
@@ -74,4 +70,61 @@ default Dimension apply(Dimension dim) {
   default boolean isPoint() {
     return false;
   }
+
+  /**
+   * Returns true if this index is a new axis, adding a dimension of size 1
+   */
+  default boolean isNewAxis() {
+    return false;
+  }
+
+  /**
+   * Returns true if this index is an ellipsis, expanding to take as many dimensions as possible (and applying all() to
+   * them)
+   */
+  default boolean isEllipsis() {
+    return false;
+  }
+
+  /**
+   * Get whether the Index supports Tensor conversion.
```

Comment:
I don't think we need to mention `for Tensor conversion` in the methods below, as long as the documentation says what the method returns, it can be of any use after that. 

Now for `tensorSupport()`, what about something more generic like `hasBoundaries()`?

Reply:
Agreed in theory, but I don't like `hasBoundaries`, because to me that sounds like it would exclude `all` and `sliceTo`, etc.  The only things that are excluded are `Hyperslab` and `Sequence`.  I went with `tensorSupport` because I couldn't think of anything better.

Maybe use "strided slice conversion" rather than "tensor conversion"?  In theory, it would apply to any library that uses strided slicing.  Or `specifiesBoundaries`.

Reply:
ok, so let's call it "strided slice conversion" and name this method `isStridedSlicingCompliant()`?